infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerCheApiInternalEnvVarProvider.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.docker;
+
+import com.google.common.base.Strings;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.provision.env.CheApiInternalEnvVarProvider;
+import org.eclipse.che.commons.lang.Pair;
+
+/**
+ * Provides env variable to docker machine with url of Che API.
+ *
+ * @author Mykhailo Kuznietsov
+ */
+@Singleton
+public class DockerCheApiInternalEnvVarProvider implements CheApiInternalEnvVarProvider {
+
+  private Pair<String, String> apiEnvVar;
+
+  @Inject
+  public DockerCheApiInternalEnvVarProvider(
+      @Named("che.infra.docker.master_api_endpoint") String apiEndpoint) {
+    String apiEndpointEnvVar = System.getenv(CHE_API_INTERNAL_VARIABLE);
```

Comment:
I would say that it should be removed. If a user wants to customize this value he is able to specify `CHE_INFRA_DOCKER_MASTER__API__ENDPOINT` environment variable.